docs/reference/analysis/tokenfilters/condition-tokenfilter.asciidoc

```diff
@@ -1,88 +1,144 @@
-                    "type" : "condition",
-                    "filter" : [ "lowercase" ],
-                    "script" : {
-                        "source" : "token.getTerm().length() < 5"  <1>
-                    }
-                }
-            }
-        }
+  "tokenizer": "standard",
+  "filter": [
+    {
+      "type": "condition",
+      "filter": [ "lowercase" ],
+      "script": {
+        "source": "token.getTerm().length() < 5"
+      }
     }
+  ],
+  "text": "THE QUICK BROWN FOX"
 }
 --------------------------------------------------
 
-<1> This will only apply the lowercase filter to terms that are less than 5
-characters in length
-
-And test it like:
+The filter produces the following tokens:
 
-[source,console]
+[source,text]
 --------------------------------------------------
-POST /condition_example/_analyze
-{
-  "analyzer" : "my_analyzer",
-  "text" : "What Flapdoodle"
-}
+[ the, QUICK, BROWN, fox ]
 --------------------------------------------------
-// TEST[continued]
-
-And it'd respond:
 
+/////////////////////
 [source,console-result]
 --------------------------------------------------
 {
-  "tokens": [
+  "tokens" : [
+    {
+      "token" : "the",
+      "start_offset" : 0,
+      "end_offset" : 3,
+      "type" : "<ALPHANUM>",
+      "position" : 0
+    },
     {
-      "token": "what",              <1>
-      "start_offset": 0,
-      "end_offset": 4,
-      "type": "<ALPHANUM>",
-      "position": 0
+      "token" : "QUICK",
+      "start_offset" : 4,
+      "end_offset" : 9,
+      "type" : "<ALPHANUM>",
+      "position" : 1
     },
     {
-      "token": "Flapdoodle",        <2>
-      "start_offset": 5,
-      "end_offset": 15,
-      "type": "<ALPHANUM>",
-      "position": 1
+      "token" : "BROWN",
+      "start_offset" : 10,
+      "end_offset" : 15,
+      "type" : "<ALPHANUM>",
+      "position" : 2
+    },
+    {
+      "token" : "fox",
+      "start_offset" : 16,
+      "end_offset" : 19,
+      "type" : "<ALPHANUM>",
+      "position" : 3
     }
   ]
 }
 --------------------------------------------------
+/////////////////////
+
+[[analysis-condition-tokenfilter-configure-parms]]
+==== Configurable parameters
+
+`filter`::
++
+--
+(Required, array of token filters)
+Array of token filters. If a token matches the predicate script in the `script`
+parameter, these filters are applied to the token in the order provided.
 
-<1> The term `What` has been lowercased, because it is only 4 characters long
-<2> The term `Flapdoodle` has been left in its original case, because it doesn't pass
-    the predicate
+These filters can include custom token filters defined in the index mapping.
+--
+
+`script`::
++
+--
+(Required, <<modules-scripting-using,script object>>)
+Predicate script used to apply token filters. If a token
+matches this script, the filters in the `filter` parameter are applied to the
+token.
+
+Only inline scripts are supported. For valid parameters, see
+<<_script_parameters>>.
```

Comment:
I'm not entirely sure where this links to, did a quick grep in the es repo and couldn't dig it up but I suppose its a general painless script page. When looking at the code about which kind of scripts are allowed here I was suprised to see that this is a specific kind of script (in the code called AnalysisPredicateScript) that seems to be limited to work on certain token attributes (like length, position etc...) only. Maybe its worth mentioning that this only works on token properties (so looks like "token.*" would always be the first part in the chain). @jdconrad should be able to validate this assumption.

Reply:
Thanks @cbuescher. For clarity, this currently links here:
https://www.elastic.co/guide/en/elasticsearch/reference/master/modules-scripting-using.html#_script_parameters

That content covers generic parameters for the script object. They aren't specific to Painless.

I think that link is still probably the best place to get those generic parameters, but I can also link to this page for Painless scripts:
https://www.elastic.co/guide/en/elasticsearch/painless/current/painless-analysis-predicate-context.html

Would that work for you @cbuescher @jdconrad? If only Painless scripts are allowed here, I can try another approach.

Reply:
Made these proposed changes in 4d127daa092 to make review a bit easier.

Reply:
Script contexts are supposed to be generic, so I would prefer to try to document them as such. I don't have a strong opinion one way or another on a link to Painless docs.